src/ApolloClient.tsx

```
@@ -1,8 +1,54 @@
-import { ApolloClient , InMemoryCache } from '@apollo/client';
+import { ApolloClient, from, HttpLink, InMemoryCache } from '@apollo/client';
+import { setContext } from '@apollo/client/link/context'
+import { onError } from '@apollo/client/link/error'
+import { jwtDecode } from 'jwt-decode'
+
+const httpLink = new HttpLink({
+  uri: process.env.REACT_APP_URL,
+})
+
+const tokenExpired = (token: string) => {
+  try {
+    const decoded: any = jwtDecode(token);
+    // console.log("Expiration time :", decoded.exp * 1000, Date.now());
+    return decoded.exp * 1000 < Date.now();
+  } catch (error) {
+    console.error("Error decoding token:", error);
```

If the token was expired, you need to throw the error, right?

---

As of for my project,just Logging out the user if the token expires